### DISCUSSION TITLE
Deprecated methods

### DISCUSSION THREAD

**Discussion by S1SYPHOS** [biblys/isbn#39] [Q&A] [answered]
Hey there,
with the new API, why would I need to validate something being ISBN-XX when I can simply convert to desired format and catch exceptions (meaning it's not valid)?

Totally curious, like this library very much :fox_face:

**Comment by clemlatz** [comment, marked as answer]
Hey @S1SYPHOS ,

Thanks for the love 🙏🏻

You’re right that, unlike with the legacy public API, using the new one, you don’t need to validate a string before you try to convert it. So instead of:

```php
$isbn = new Isbn("9781234567890");
if ($isbn->isValid) {
  $isbn13 = $isbn->format("ISBN-13");
}
```

Now you can just do:

```php
$isbn13 = Isbn::convertToIsbn13("9781234567890");
```

and catch exceptions. 

But if you don’t need to convert a string and you only need to validate it, you can call:

```php
$isbn13 = Isbn::validateAsIsbn13("9781234567890 ");
```

and catch exceptions. 

(I trust that you already understood all this, but I thought this might help other users that may be confused about the new API). 

The difference between those two methods is that `convert…` is less strict and will only throw if the string passed as an argument is absolutely unparsable, while `validate…` will throw even for a simple checksum character difference. In the legacy API, some users were confused that `isValid` would return true and `validate` would not throw for strings that were not stricly valid ISBNs because they were still parsable.

While you're here, as you asked in a previous issue, you'll be happy to know that I plan to bring real exceptions to the library so you can distinguish between parsing (unrecoverable) and validation errors, with even specific exceptions so you can programmatically react depending on the reason why parsing or validation failed.

**Reply by S1SYPHOS** [reply]
LOVE IT!

Thanks for the detailed explanation. I'll go with converting only then for my project, that should be sufficient!
